src/V3EmitMkJson.cpp

```diff
@@ -0,0 +1,320 @@
+// -*- mode: C++; c-file-style: "cc-mode" -*-
+//*************************************************************************
+// DESCRIPTION: Verilator: Emit JSON manifest file
+//
+// Code available from: https://verilator.org
+//
+//*************************************************************************
+//
+// Copyright 2004-2025 by Wilson Snyder. This program is free software; you
+// can redistribute it and/or modify it under the terms of either the GNU
+// Lesser General Public License Version 3 or the Perl Artistic License
+// Version 2.0.
+// SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0
+//
+//*************************************************************************
+
+#include "V3PchAstNoMT.h"  // VL_MT_DISABLED_CODE_UNIT
+
+#include "V3EmitMkJson.h"
+
+#include "V3EmitCBase.h"
+#include "V3HierBlock.h"
+#include "V3Os.h"
+
+#include <memory>
+
+VL_DEFINE_DEBUG_FUNCTIONS;
+
+// ######################################################################
+//  Emit statements
+
+class V3EmitMkJsonEmitter final {
+    class Printer final {
+        // MEMBERS
+    private:
+        const std::unique_ptr<std::ofstream>& m_of;
+        std::stack<char> m_scope;  // stack of ']' and '}' to be used to close currently open scopes
+        std::string m_prefix;  // prefix emited before each line in the current scope (indent * scope depth)
+        std::string m_indent;  // single indent
+        bool m_empty = true;  // indicates if the current scope is empty
+
+        // METHODS
+    public:
+        Printer(const std::unique_ptr<std::ofstream>& of, const std::string& indent = "    ")
+            : m_of(of)
+            , m_indent(indent) {
+            begin();
+        }
+
+        ~Printer() { end(); }
+
+        Printer& begin(const std::string& name, char type = '{') {
+            if (!m_empty) { *m_of << ",\n"; }
```

Comment:
```suggestion
            if (!m_empty++) *m_of << ",\n";
```

Make m_empty an int, then this trick means you don't need to clear m_empty in those places where you do right below this. Similar others.

Reply:
Good point.

Reply:
Actually it makes code more ugly. I reverted back (the condition has to be (m_fieldsCount++ > 1) with initial setting to -1 which is counter intuitive. And it saves only 6 lines. I won't do that.